tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1370,6 +1370,305 @@ static void test_rx_priv_interleave(void)
 
 	PASS();
 }
+
+#if CONFIG_BT_MESH_GATT_PROXY
+
+#define BEACON_TYPE_PRIVATE_NET_ID  2
+#define BEACON_TYPE_PRIVATE_NODE_ID 3
+#define BEACON_TYPE_PRIVATE_LEN     28
+
+static const uint8_t test_netkey_idx; /*It is 0 but compliance won't let me emphasize that*/
+static const uint8_t test_netkey_sec_idx = 1;
+static uint64_t random_field;
+
+static bool pp_type_check(uint16_t expected_beacon, uint8_t adv_type, struct net_buf_simple *buf)
+{
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND || buf->len != BEACON_TYPE_PRIVATE_LEN) {
+		return false;
+	}
+
+	/* Remove Header */
+	(void)net_buf_simple_pull_mem(buf, 11);
+
+	uint8_t beacon_type = net_buf_simple_pull_u8(buf);
+
+	if (beacon_type != expected_beacon) {
+		return false;
+	}
+
+	return true;
+}
+
+static uint64_t pp_hash_calc(const uint8_t netkey[16], uint64_t random, uint16_t *addr)
+{
+	uint64_t hash;
+	uint8_t tmp[16] = {0, 0, 0, 0, 0, 3};
+	struct bt_mesh_key id_key;
+
+	bt_mesh_identity_key(netkey, &id_key);
+
+	if (addr) {
+		memcpy(&tmp[6], &random, 8);
+		sys_put_be16(*addr, &tmp[14]);
+
+	} else {
+		bt_mesh_k3(netkey, &tmp[0]);
+		memcpy(&tmp[8], &random, 8);
+	}
+
+	bt_mesh_encrypt(&id_key, tmp, tmp);
+	memcpy(&hash, &tmp[8], 8);
+
+	return hash;
+}
+
+static void priv_net_id_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				struct net_buf_simple *buf)
+{
+	if (!pp_type_check(BEACON_TYPE_PRIVATE_NET_ID, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	uint64_t hash = net_buf_simple_pull_le64(buf);
+
+	random_field = net_buf_simple_pull_le64(buf);
+	ASSERT_EQUAL(hash, pp_hash_calc(test_net_key, random_field, NULL));
+
+	k_sem_give(&observer_sem);
+}
+
+static void priv_node_id_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				 struct net_buf_simple *buf)
+{
+	if (!pp_type_check(BEACON_TYPE_PRIVATE_NODE_ID, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	uint64_t hash = net_buf_simple_pull_le64(buf);
+
+	random_field = net_buf_simple_pull_le64(buf);
+	ASSERT_EQUAL(hash, pp_hash_calc(test_net_key, random_field, (uint16_t *)&tx_cfg.addr));
+
+	k_sem_give(&observer_sem);
+}
+
+static void priv_multi_net_id_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				      struct net_buf_simple *buf)
+{
+	static bool net1_rcv, net2_rcv;
+
+	if (!pp_type_check(BEACON_TYPE_PRIVATE_NET_ID, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	uint64_t hash = net_buf_simple_pull_le64(buf);
+
+	random_field = net_buf_simple_pull_le64(buf);
+
+	if (hash == pp_hash_calc(test_net_key, random_field, NULL)) {
+		net1_rcv = true;
+	} else if (hash == pp_hash_calc(test_net_key_secondary, random_field, NULL)) {
+		net2_rcv = true;
+	}
+
+	/* Both networks has been received*/
+	if (net1_rcv && net2_rcv) {
+		k_sem_give(&observer_sem);
+	}
+}
+
+static bool wait_for_priv_beacon(bt_le_scan_cb_t cb, uint16_t wait_ms)
```

Comment:
It looks identical to `wait_fot_beacon` but has timeout as an argument, while `wait_for_beacon` has it hardcoded. Perhaps they can be merged?